core/src/main/java/org/apache/druid/data/input/IntermediateRowParsingReader.java

```diff
@@ -96,23 +101,45 @@ public void close() throws IOException
   public CloseableIterator<InputRowListPlusRawValues> sample() throws IOException
   {
     return intermediateRowIterator().map(row -> {
-      final Map<String, Object> rawColumns;
+
+      final List<Map<String, Object>> rawColumnsList;
       try {
-        rawColumns = toMap(row);
+        rawColumnsList = toMap(row);
       }
       catch (Exception e) {
-        return InputRowListPlusRawValues.of(null, new ParseException(e, "Unable to parse row [%s] into JSON", row));
+        return Collections.singletonList(InputRowListPlusRawValues.of(null,
+                                                                      new ParseException(e, "Unable to parse row [%s] into JSON", row)));
+      }
+
+      if (CollectionUtils.isNullOrEmpty(rawColumnsList)) {
+        return Collections.singletonList(InputRowListPlusRawValues.of(null,
+                                                                      new ParseException("No map object parsed for row [%s]", row)));
       }
+
+      List<InputRow> rows;
       try {
-        return InputRowListPlusRawValues.of(parseInputRows(row), rawColumns);
+        rows = parseInputRows(row);
       }
       catch (ParseException e) {
-        return InputRowListPlusRawValues.of(rawColumns, e);
+        return rawColumnsList.stream().map(rawColumn -> InputRowListPlusRawValues.of(rawColumn, e)).collect(Collectors.toList());
       }
       catch (IOException e) {
-        return InputRowListPlusRawValues.of(rawColumns, new ParseException(e, "Unable to parse row [%s] into inputRow", row));
+        ParseException exception = new ParseException(e, "Unable to parse row [%s] into inputRow", row);
+        return rawColumnsList.stream().map(rawColumn -> InputRowListPlusRawValues.of(rawColumn, exception)).collect(Collectors.toList());
+      }
```

Comment:
This doesn't still seem correct to me.. The purpose of returning `rawColumn` and `exception` in `InputRowListPlusRawValues` is showing them in the sampler altogether, so that users will debug their data and input format easily. To do that, we should show exactly what the data was when a `ParseException` is thrown. Now, thinking about the new behavior parsing a list of JSONs into `InputRow`s, `row` will contain the list of JSONs here. If a `ParseException` was thrown while parsing one of them, we should return one `InputRowListPlusRawValues` which contains `row` (the whole list of JSONs) and the exception because we don't know exactly in which JSON in the list the `ParseException` was thrown from. For this, I still think we should change `rawValues` in `InputRowListPlusRawValues` to be `List<Map<String, Object>>`. I understand that you don't want to touch it as it is widely used in unit tests, but I'm not sure if we can fix this without touching it.

Reply:
I agree with you. 

`rawValues` in `InputRowListPlusRawValues` should also be a list to correspond the list of `inputRows`. Since the `rawValues` is also passed to `SamplerResponseRow` when exception occurs, to change `rawValues` from `Map` to `List<Map>` also involves changes of the web console. I don't know if the change of front-end should also be included in this PR.

Reply:
Good point. I'm not sure if we have a unit test for parsing sampler response. If we have, the CI will fail and we need to fix it in this PR together. Otherwise, I'm OK with doing the fix in a follow-up. @vogievetsky do you know whether or not we have such a unit test?